bibliothecarius-service/src/main/java/com/peony/bibliothecarius/service/application/service/FileService.java

```diff
@@ -0,0 +1,85 @@
+
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectInputStream;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.util.IOUtils;
+import com.peony.bibliothecarius.service.config.S3Config;
+import com.peony.bibliothecarius.service.infrastructure.rest.exception.FileUploadFailed;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class FileService {
+
+    public static final String ERROR_TEMPLATE = "File upload failed";
+
+    private final S3Config s3Config;
+    private  final AmazonS3 s3;
+
+    public String saveFile(MultipartFile file) {
+        String originalFilename = file.getOriginalFilename();
+        try {
+            File file1 = convertMultiPartToFile(file);
+            PutObjectResult putObjectResult = s3.putObject(s3Config.getBucketName(), originalFilename, file1);
+            return putObjectResult.getContentMd5();
+        } catch (IOException e) {
+            log.error("File '{}' upload failed", file);
+            throw new FileUploadFailed(String.format(ERROR_TEMPLATE));
+        }
+    }
+
+    public byte[] downloadFile(String filename) {
+        S3Object object = s3.getObject(s3Config.getBucketName(), filename);
+        S3ObjectInputStream objectContent = object.getObjectContent();
+        try {
+            return IOUtils.toByteArray(objectContent);
+        } catch (IOException e) {
+            log.error("File '{}' download failed", filename);
+            throw new FileUploadFailed(String.format(ERROR_TEMPLATE));
+        }
+
+    }
+
+    public String deleteFile(String filename) {
+        if (!s3.doesBucketExistV2(filename)) {
+            log.error("File '{}' does not exist", filename);
+            throw new FileUploadFailed(String.format(ERROR_TEMPLATE));
+        }
+        s3.deleteObject(s3Config.getBucketName(), filename);
+        return "File deleted";
+    }
+
+    public List<String> listAllFiles() {
+
+        ListObjectsV2Result listObjectsV2Result = s3.listObjectsV2(s3Config.getBucketName());
+        return  listObjectsV2Result.getObjectSummaries()
+                .stream()
+                .map(S3ObjectSummary::getKey)
+                .collect(Collectors.toList());
+
+    }
+
+    private File convertMultiPartToFile(MultipartFile file) throws IOException {
+        File convFile = new File(file.getOriginalFilename());
+        FileOutputStream fos = new FileOutputStream(convFile);
+        fos.write(file.getBytes());
+        fos.close();
```

Comment:
```suggestion
        try(FileOutputStream fos = new FileOutputStream(convFile)) {
                fos.write(file.getBytes());
        }
```
this code construction is called try-with-resources, you could pass objects if classes that implement Closeable interface into () aftre try and them when try block ends close method will be called automatically